app/src/main/java/com/github/se/eduverse/viewmodel/PublicationViewModel.kt

```diff
@@ -58,4 +62,21 @@ open class PublicationViewModel(
       }
     }
   }
+
+  open suspend fun loadFollowedPublications(userIds: List<String> = emptyList()) {
+    viewModelScope.launch {
+      try {
+        if (_followedPublications.value.isEmpty()) {
+          val newPublications = repository.loadRandomPublications(userIds)
+          _followedPublications.value = newPublications
+        } else {
+          val morePublications = repository.loadRandomPublications(userIds)
+          _followedPublications.value = (_followedPublications.value + morePublications).shuffled()
+        }
+        _error.value = null
+      } catch (e: Exception) {
+        _error.value = "fail to load publications"
+      }
+    }
+  }
```

Comment:
The function correctly updates the UI state by fetching new publications or appending additional ones when more are loaded. However, since it’s already using viewModelScope.launch, it might not need to be marked as a suspend function. Additionally, you could consider introducing a loading state or using a try-catch block outside of the launched coroutine to ensure the calling context handles errors and loading states more clearly.